### PR TITLE
DM-39781: Clean docs and add ruff configuration

### PR DESCRIPTION
**Pull request by timj**
This ruff configuration matches the flake8 and pydocstyle configurations already in place but fixes all the docstrings to enable ruff to run on the entire repository.

## Checklist

- [x] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/daf/butler/cli/butler.py [hunk 1]
```diff
@@ -123,9 +126,11 @@ def getLocalCommands(self) -> defaultdict[str, list[str]]:
         )
 
     def list_commands(self, ctx: click.Context) -> list[str]:
-        """Used by Click to get all the commands that can be called by the
+        """Gget all the commands that can be called by the
```
Comment (ktlim) [inline python/lsst/daf/butler/cli/butler.py]:
```suggestion
        """Get all the commands that can be called by the
```
:)

File: pyproject.toml [hunk 2]
```diff
@@ -151,3 +151,46 @@ exclude_lines = [
...
+    "D100",
+    "D200",
+    "D205",
+    "D400",
+    "E402", # Module level import not at top of file (against Rubin style)
+]
+line-length = 110
+select = [
+    "E",  # pycodestyle
+    "F",  # pydodestyle
```
Comment (ktlim) [inline pyproject.toml]:
```suggestion
    "F",  # pycodestyle
```

File: pyproject.toml [hunk 3]
```diff
@@ -151,3 +151,49 @@ exclude_lines = [
...
+    "W",  # pycodestyle
+    "D",  # pydocstyle
+]
+target-version = "py310"
+extend-select = [
+    "RUF100", # Warn about unused noqa
+]
+
+[tool.ruff.per-file-ignores]
+"python/lsst/daf/butler/registry/queries/expressions/parser/parserYacc.py" = ["D401", "D403"]
```
Comment (andy-slac) [inline pyproject.toml:194]: This file uses docstrings for grammar description, they cannot be "fixed".

File: tests/test_query_relations.py [hunk 4]
```diff
@@ -95,7 +95,7 @@ def assert_relation_str(
         | queries.DimensionRecordQueryResults
         | queries.ParentDatasetQueryResults,
     ) -> None:
-        """A specialized test assert that checks that one or more registry
+        """Assert that checks that one or more registry
```
Comment (andy-slac) [inline tests/test_query_relations.py:98]: Does it think that "Assert" is a verb here? 🙂

**Review by andy-slac — APPROVED** [review]
Looks good.

**Pull request merged**